egs/librispeech/ASR/transducer_stateless/decoder.py

```diff
@@ -82,17 +84,30 @@ def forward(self, y: torch.Tensor, need_pad: bool = True) -> torch.Tensor:
         Returns:
           Return a tensor of shape (N, U, embedding_dim).
         """
-        embeding_out = self.embedding(y)
+        embedding_out = self.embedding(y)
         if self.context_size > 1:
-            embeding_out = embeding_out.permute(0, 2, 1)
+            embedding_out = embedding_out.permute(0, 2, 1)
             if need_pad is True:
-                embeding_out = F.pad(
-                    embeding_out, pad=(self.context_size - 1, 0)
-                )
+                # If the input is [sos, a, b, c, d] and output is
```

Comment:
This  comment is not very clear.  When you say "input" and "output", please clarify what variables you are talking about.  It would be nice to show what the arrays are supposed to look like after padding.